content/influxdb/cloud-iox/sql-reference.md

```diff
@@ -0,0 +1,614 @@
+| COUNT()  | Returns returns the number of rows from a field or tag key |
+| AVG()    | Returns the average value of a column                      |
+| SUM()    | Returns the summed value of a column                       |
+| MEAN()   | Returns the mean value of a column                         |
+| MIN()    | Returns the smallest value of the selected column          |
+| MAX()    | Returns the largest value of the selected column          |
+
+#### Examples
+
+```sql
+
+SELECT COUNT("water_level") 
+FROM "h2o_feet"
+
+SELECT AVG("water_level"), "location"
+FROM "h2o_feet" 
+GROUP BY "location"
+
+SELECT SUM("water_level"), "location"
+FROM "h2o_feet" 
+GROUP BY "location"
+```
+
+### Selector functions
+
+Selector functions are unique to InfluxDB. They behave like aggregate functions in that they take a row of data and compute it down to a single value.  However, selectors are unique in that they return a **time value** in addition to the computed value. In short, selectors retrun an aggreagetd value along with a timestamp. 
+
+| Function         | Description                                                     |
+| :--------------- | :-------------------------------------------------------------- |
+| SELECTOR_FIRST() | Returns the first value of a selected column and timestamp.   |
+| SELECTOR_LAST()  | Returns the last value of a selected column and timestamp.     |
+| SELECTOR_MIN()   | Returns the smallest value of a selected column and timestamp. |
+| SELECTOR_MAX()   | Returns the largest value of a selected column and timestamp.  |
+
+#### Examples
+
+```sql
+SELECT 
+SELECTOR_MAX("pH", time)['value'],
+SELECTOR_MAX("pH", time)['time']
+FROM "h2o_pH"
+
+SELECT 
+SELECTOR_LAST("water_level", time)['value'],
+SELECTOR_LAST("water_level", time)['time']
+FROM "h2o_feet"
+WHERE time >= timestamp '2019-09-10T00:00:00Z' AND time <= timestamp '2019-09-19T00:00:00Z'
+
+```
+
+### Time series functions
+
+| Function              | Description                                                                                     |
+| :-------------------- | :---------------------------------------------------------------------------------------------- |
+| TIME_BUCKET_GAPFILL() | Returns a contiguous set of time bucketed data.                                                 |
+| DATEBIN()             | Bins the input timestamp into a specified interval.                                             |
+| DATE_TRUNC()          | Truncates a timestamp expression based on the date part specified, such as hour, day, or month. |
+| DATE_PART()           | Returns the specified part of a date.                                                           |
+| NOW()                 | Returns the current time.                                                                       |
+|                       
+                 
+#### Examples
+
+```sql
+SELECT time_bucket_gapfill('1 day', time) as day,
```

Comment:
This function is not yet completely implemented -- I am not sure what our policy on documenting "in the works" features is.  @wolffcm  may be able to give you more context

Reply:
I have begun implementing a function called `DATE_BIN_GAPFILL` that takes the same arguments as `DATE_BIN` but fills gaps. At the moment it will return a `not implemented` error if you try to use it.

We will also be providing a `TIME_BUCKET_GAPFILL` function that is an alias for `DATE_BIN_GAPFILL`, for users who migrate from the other platforms. So, long story short, neither of those functions are ready for users yet.

As far as documenting in-the-works-features, we tended not to for Flux, to avoid people being confused/disappointed if they try something that was documented but did not work.

Reply:
We generally don't add features or things "in the works" because it does cause confusion.  Do you think this will most likely be implemented after the Jan 31 release?  If so I will remove it for now.  I can alway add it back once it's ready, that's easy to do.

Reply:
Yes, it will be implemented after Jan 31. Best to remove it for now.

Reply:
removed for now